src/Libraries/CoreNodeModelsWpf/NodeViewCustomizations/Watch.cs

```diff
@@ -43,20 +43,35 @@ public void CustomizeView(Watch nodeModel, NodeView nodeView)
             rootWatchViewModel = new WatchViewModel(dynamoViewModel.BackgroundPreviewViewModel.AddLabelForPath);
 
             var watchTree = new WatchTree(rootWatchViewModel);
+
+            rootWatchViewModel.WatchNode = watch;
             watchTree.BorderThickness = new Thickness(1, 1, 1, 1);
             watchTree.BorderBrush = new SolidColorBrush(Color.FromRgb(220,220,220));
 
             watchTree.SetWatchNodeProperties();
 
+            nodeView.SizeChanged += (sender, args) =>
+                nodeModel.SetSize(args.NewSize.Width, args.NewSize.Height);
```

Comment:
Can you check how anonymous handler got unsubscribed? I remember they won't be unhandled correctly in certain cases, just want to make sure